app/src/main/scala/org/alephium/explorer/persistence/queries/BlockDepQueries.scala

```diff
@@ -0,0 +1,68 @@
+// Copyright 2018 The Alephium Authors
+// This file is part of the alephium project.
+//
+// The library is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Lesser General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// The library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with the library. If not, see <http://www.gnu.org/licenses/>.
+
+package org.alephium.explorer.persistence.queries
+
+import slick.dbio.DBIOAction
+import slick.jdbc.{PositionedParameters, SetParameter, SQLActionBuilder}
+
+import org.alephium.explorer.persistence.DBActionW
+import org.alephium.explorer.persistence.model.BlockDepEntity
+import org.alephium.explorer.persistence.schema.CustomSetParameter._
+
+object BlockDepQueries {
+
+  /**
+    * Insert block_deps or update if there is a primary key conflict.
```

Comment:
This is a SQL translation of how the code works currently. Existing code performs [`insertOrUpdate`](https://github.com/alephium/explorer-backend/blob/master/app/src/main/scala/org/alephium/explorer/persistence/queries/BlockQueries.scala#L87).

Reply:
when we start syncing, we always step back few blocks to be sure we don't miss some blocks or new forks, so we do re-insert blocks, we could maybe also just do an `ON CONFLICT IGNORE` in case on primary key conflict